common/src/main/scala/net/psforever/objects/avatar/PlayerControl.scala

```diff
@@ -877,13 +880,16 @@ object PlayerControl {
     * na
     * @param target na
     */
-  def HandleDamage(target : Player, cause : ResolvedProjectile, damageToHealth : Int, damageToArmor : Int, damageToCapacitor : Int) : Unit = {
+  def HandleDamage(target : Player, cause : ResolvedProjectile, damageToHealth : Int, damageToArmor : Int, damageToStamina : Int, damageToCapacitor : Int) : Unit = {
     val targetGUID = target.GUID
     val zone = target.Zone
     val zoneId = zone.Id
     val events = zone.AvatarEvents
     val health = target.Health
     if(health > 0) {
+      if(damageToStamina > 0) {
+        target.Actor ! Player.DrainStamina(0)
```

Comment:
Are we passing 0 to force a stamina update to be sent to the client? If so, could we not move the actual stamina drain from `ResolutionCalculations` to here?

Reply:
Actually, since this calls `StaminaChanged` that has a check to make sure the stamina has changed before calling `UpdateStamina`, so it's doing nothing?

Reply:
I had thought so too but then I figured you probably have a good reason to want to update the value of stamina exactly when you do in `ResoCalc`.

Reply:
Nothing more than it seemed like a logical place to put it and it worked, since the `Stamina` setter would call out to `PlayerControl` no matter where the stamina was updated

Reply:
In any event, we should be able to modify entity fields without forcing client updates.